pkgs/by-name/st/structurizr-cli/package.nix

```
@@ -0,0 +1,61 @@
+{
+  lib,
+  stdenv,
+  fetchzip,
+  makeBinaryWrapper,
+  jre,
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "structurizr-cli";
+  version = "2025.05.28";
+
+  src = fetchzip {
+    url = "https://github.com/structurizr/cli/releases/download/v${finalAttrs.version}/structurizr-cli.zip";
+    hash = "sha256-UYqUZydjsAoy5be/UhAyX/7OvLq8pXA6STwbEnCG7CU=";
+    stripRoot = false;
+  };
+
+  strictDeps = true;
+  nativeBuildInputs = [ makeBinaryWrapper ];
+  buildInputs = [ jre ];
+
+  installPhase = ''
+    runHook preInstall
+
+    # Create folder structure
+    mkdir -p $out/bin
+    # Copy all jars
+    cp -r $src/. $out/
+
+    # Find the names of all jars
+    jars=$(find $out/lib -name "*.jar" | tr '\n' ':')
+    jars=$(echo $jars | sed 's/:$//')
+
+    # Create custom wrapper, since we want to inject our own java
+    makeBinaryWrapper ${jre}/bin/java $out/bin/structurizr-cli \
+      --add-flags "-cp $jars com.structurizr.cli.StructurizrCliApplication"
+
+    runHook postInstall
+  '';
+
+  doInstallCheck = true;
+
+  installCheckPhase = ''
+    $out/bin/structurizr-cli help > /dev/null
+  '';
+
+  versionCheckHook = ''
+    $out/bin/structurizr-cli version | grep -q "structurizr-cli: ${finalAttrs.version}"
+  '';
```

https://github.com/NixOS/nixpkgs/blob/master/doc/hooks/versionCheckHook.section.md

```suggestion
  doInstallCheck = true;
    nativeInstallCheckInputs = [
    versionCheckHook
  ];
  versionCheckProgramArg = "version";
```